src/lib/react/wrap/ReactWrapperMacro.hx

```diff
@@ -0,0 +1,59 @@
+package react.wrap;
+
+import haxe.macro.ComplexTypeTools;
+import haxe.macro.Context;
+import haxe.macro.Expr;
+import haxe.macro.ExprTools;
+import haxe.macro.Type;
+import haxe.macro.TypeTools;
+import react.jsx.JsxStaticMacro;
+
+class ReactWrapperMacro
+{
+	static public inline var WRAP_META = ':wrap';
+	static inline var WRAPPED_META = ':wrapped_by_macro';
+
+	static public function buildComponent(inClass:ClassType, fields:Array<Field>):Array<Field>
+	{
+		if (inClass.meta.has(WRAPPED_META)) return fields;
```

Comment:
Really cool, but is it possible for multiple builds to happen?

Reply:
With compilation server, if the component (or its meta) is modified it looses all macro-generated meta and its build macros are re-run.